### PR TITLE
ci: add annotation for linters

### PR DESCRIPTION
**Pull request by davejrt**
Adds the `sg lint --fix` to an annotation above the builds so we can see them more clearly

## Test plan

image

### CODE REVIEW

File: dev/sg/linters/runner.go [hunk 1]
```diff
@@ -18,6 +20,17 @@ func NewRunner(out *std.Output, annotations bool, targets ...Target) Runner {
 		if c.Fix == nil {
 			return ""
 		}
+		if annotations {
+			path := fmt.Sprintf("../../%s.md", category)
+			of, err := os.Create(path)
```
Comment (burmudar) [inline dev/sg/linters/runner.go:25]: minor nit 
```suggestion
			fd, err := os.Create(path)
```

File: dev/sg/linters/runner.go [hunk 2]
```diff
@@ -18,6 +20,17 @@ func NewRunner(out *std.Output, annotations bool, targets ...Target) Runner {
...
+			if err != nil {
+				log.Print(err)
```
Comment (burmudar) [inline dev/sg/linters/runner.go:27]: Should we just write to `os.Stderr` here instead 🤔

Comment (davejrt) [reply dev/sg/linters/runner.go:27]: changed in favor of os.stderr to avoid importing log for a one liner where we can already used os

**Comment by burmudar** [comment]
For the second annotation, it is possible to trim the output? Since we already have the output from go generate above, we would be annotating twice the same output?

**Comment by davejrt** [comment]
@burmudar nice idea...but I'm not entirely sure how, or how I'd test it locally. Is there another example where we aggregate this? 

Or is the suggestion that the second error like in the image above not get shown at all?

**Comment by burmudar** [comment]
> @burmudar nice idea...but I'm not entirely sure how, or how I'd test it locally. Is there another example where we aggregate this? 
> 
> 
> 
> Or is the suggestion that the second error like in the image above not get shown at all? 

@davejrt yeah, I think the second error shouldn't be shown. It just adds noise I think

**Comment by davejrt** [comment]
@burmudar I think the issue with only showing the first check in this case, is these errors are from two different steps. Unless we write logic in the upload to ignore any other annotations it's probably a bit much given this is a stop gap in the grander scheme of bazel. Below is a new output that at least keeps the gofmt and go generate errors together, rather then splitting them up. 

image

File: dev/sg/internal/check/annotation.go [hunk 3]
```diff
@@ -21,4 +21,38 @@ func generateAnnotation(category string, check string, content string) {
 	// write annotation
 	path := filepath.Join(annotationsDir, fmt.Sprintf("%s: %s.md", category, check))
 	_ = os.WriteFile(path, []byte(content+"\n"), os.ModePerm)
+
+	if check == "Go format" {
+		gofmt, _ := os.Open(fmt.Sprintf("%s/gofmt", annotationsDir))
+		if err != nil {
+			os.Stderr.WriteString(err.Error() + "\n")
```
Comment (burmudar) [inline dev/sg/internal/check/annotation.go:28]: Might be better to use `fmt.Fprintln(os.Stderr, err.Error())`

**Review by burmudar — APPROVED** [review]
One small comments but overall 🆗

File: dev/sg/internal/check/annotation.go (hunk 3 above)
Comment (davejrt) [reply dev/sg/internal/check/annotation.go:28]: does os.Stderr create an exit code 1 after it runs?

Comment (burmudar) [reply dev/sg/internal/check/annotation.go:28]: Nope, it's just a file at the end of the day https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/os/file.go;l=64

**Pull request merged**